### PR TITLE
doc/mgr: Add cli api documentation

### PR DESCRIPTION
**Pull request by Waadkh7**
Fixes: https://tracker.ceph.com/issues/52120
Signed-off-by: Waad AlKhoury <walkhour@redhat.com>
Depends on #42436

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/mgr/cli_api_commands.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Commands
+--------
+
+To execute the module, run::
+
+  ceph mgr cli <api> <param>
+
+To execute the module benchmark, run::
+
+  ceph mgr cli benchmark <api> <param> <number of calls> <number of threads>
```
Comment (pereman2) [inline doc/mgr/cli_api_commands.rst:26]:
```suggestion
To run a mgr module command, run::

  ceph mgr cli <command> <param>

List all available mgr module commands with , run::

  ceph mgr cli --help

To benchmark a command, run::

  ceph mgr cli benchmark <command> <param> <number of calls> <number of threads>
```

File: doc/mgr/cli_api_commands.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
+CLI API Commands Module
+=======================
+
+The CLI API module expose most ceph-mgr python API via CLI and benchmarking them.
```
Comment (pereman2) [inline doc/mgr/cli_api_commands.rst:4]:
```suggestion
The CLI API module exposes most ceph-mgr python API via CLI. Furthermore, this API can be benchmarked for further testing.
```

**Review by pereman2 — COMMENTED** [review]
nits

**Review by alfonsomthd — APPROVED** [review]

File: doc/mgr/cli_api_commands.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+The CLI API module exposes most ceph-mgr python API via CLI. Furthermore, this API can be
+benchmarked for further testing.
+
+Enabling
+--------
+
+The *cli api commands* module is enabled with::
+
+  ceph mgr module enable cli_api_commands
```
Comment (alfonsomthd) [inline doc/mgr/cli_api_commands.rst:12]:
```suggestion
  ceph mgr module enable cli_api
```

File: doc/mgr/cli_api_commands.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+To check that it is enabled, run::
+
+  ceph mgr module ls
```
Comment (alfonsomthd) [inline doc/mgr/cli_api_commands.rst:16]:
```suggestion
  ceph mgr module ls | grep cli_api
```

File: doc/mgr/index.rst [hunk 5]
```diff
@@ -48,3 +48,4 @@ sensible.
     MDS Autoscaler module <mds_autoscaler>
     NFS module <nfs>
     Progress Module <progress>
+    CLI API Commands module <cli_api_commands>
```
Comment (alfonsomthd) [inline doc/mgr/index.rst:51]:
```suggestion
    CLI API Commands module <cli_api>
```

**Review by alfonsomthd — CHANGES_REQUESTED** [review]

**Pull request merged**